apps/wallet/src/ui/app/staking/stake/StakeForm.tsx

```diff
@@ -46,48 +50,107 @@ function StakeForm({
 
     const [formatted, symbol] = useFormatCoin(coinBalance, coinType);
 
+    const [coinDecimals] = useCoinDecimals(coinType);
+
+    const [gasBudgetEstimation] = useFormatCoin(
+        DEFAULT_GAS_BUDGET_FOR_STAKE,
+        coinType
```

Comment:
It seems that the component assumes `coinType` can be any coin type. But Gas is always SUI, we should use `SUI_TYPE_ARG`